lib/project_types/extension/models/types/checkout_post_purchase.rb

```diff
@@ -12,7 +12,10 @@ def create(directory_name, context)
         end
 
         def config(context)
-          Features::Argo.checkout.config(context)
+          {
+            **Features::Argo.checkout.config(context),
+            **Features::ArgoConfig.parse_yaml(context),
```

Comment:
Another issue is that as we discussed earlier this week, we do not yet have checks in place on the core side to be able to handle keys in the config that are unrecognized. The `ValidateVersion` logic blows up. We should probably have checks on one side or the other before releasing this code. The reason these checks do not currently exist is because in all current flows we dictate what the config payload is shaped like (in Partners).

Reply:
> Perhaps for now we could switch them

Good idea, will change that. 

> Also, how does the user of the cli know what the valid config keys are? Docs?

Yes, docs for now. `metafields` is the only entry currently.

> but maybe in the long run we should consider having this payload conform to a schema

I do absolutely agree. Not sure if that'd be something that could even be shared in cli and core. 

> The reason these checks do not currently exist is because in all current flows we dictate what the config payload is shaped like (in Partners).

I'd assume that we'd want to do that in core eventually even for the dictated payload as folks can fabricate any http request and cause a 50X response in core. 
For the time being, we could throw an error when any non-allowed keys are found from within the CLI. The validation of allowed keys (metafields) itself is being implemented in a PR in core.

The initial version of this PR already filtered unpermitted keys, but did so silently. I do believe we should tell users about these sorts of errors, though.

Reply:
@jantnovi I added a small key filter in https://github.com/Shopify/shopify-app-cli/pull/857/commits/74c3b0155034927cfc981052afd866b345206c3b 

it aborts when unpermitted keys are found in the yml. This could be exchanged with a full-on per-extension schema validation in the future. Actual validation happens on the receiving end regardless.

what do you think?

Reply:
> I would assume that we would want to do that in core eventually even for the dictated payload as folks can fabricate any http request and cause a 50X response in core.

Ya they could for sure already do this. This just makes it a lot easier, and makes it possible to do accidentally. I've got an issue open to handle unknown keys here: https://github.com/Shopify/app-extension-libs/issues/898

I think having this quick check against the keys mitigates that issue. We will have to come up with a more formal solution for this at some point, but this works for me for now.